src/lib.rs

```diff
@@ -512,27 +512,54 @@ pub(crate) fn warn_host_version_mismatch(
     Ok(VersionMatch::Same)
 }
 
-/// Parses the `Cross.toml` at the root of the Cargo project or from the
-/// `CROSS_CONFIG` environment variable (if any exist in either location).
+/// Obtains the [`CrossToml`] from one of the possible locations
+///
+/// These locations are checked in the following order:
+/// 1. Package metadata in the Cargo.toml
+/// 2. If the `CROSS_CONFIG` variable is set, it tries to read the config from its value
+/// 3. Otherwise, the `Cross.toml` in the project root is set.
```

Comment:
I believe we're changing this order, see https://github.com/cross-rs/cross/pull/754#issuecomment-1151546469.

This way, `Cross.toml` has the highest priority, and overrides any settings provided in `Cargo.toml`. So it would be `CROSS_CONFIG` > `Cross.toml` > `Cargo.toml`.